base/regex.jl

```diff
@@ -243,15 +245,6 @@ end
 
 eachmatch(re::Regex, str::AbstractString) = RegexMatchIterator(re,str)
 
-# miscellaneous methods that depend on Regex being defined
-
-filter!(r::Regex, v) = filter!(x->ismatch(r,x), v)
-filter(r::Regex, v)  = filter(x->ismatch(r,x), v)
-
-filter!(r::Regex, d::Dict) = filter!((k,v)->ismatch(r,k),d)
```

Comment:
Technically this would need a deprecation, since filter on Dicts expects a 2-argument function. Or I guess we could add a `call` overload for `Regex` with 2 extra arguments, but that seems too magical since it really only makes sense for this use case.